ipynb/primary-processing/11-generate_per_bodysite_results.md

```diff
@@ -0,0 +1,278 @@
+...
+...     # Makes the new directory
+...     !mkdir -p $sink_dir
+...
+...     # Copies the files over
+...     !cp $source_otu $sink_otu
+...     !cp $source_map $sink_map
+```
+
+We'll follow by moving over the files for all participants. We're going to have a mapping file including alpha diversity, a rarefied OTU table, and distance matrices for each rarefaction depth. Once again, we'll build a helper function to find our file names, and then move them over.
+
+```python
+>>> def rarefied_parameter_iterator():
+...     for bodysite in ['fecal', 'oral', 'skin']:
+...         for trim in ['notrim', '100nt']:
+...             for depth_id, depth in enumerate(['1k', '10k']):
+...                 source_otu = '"%s"' % agu.get_existing_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['source-%s-otu' % depth])
+...                 source_map = agu.get_existing_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['source-%s-map' % depth])
+...
+...                 source_alpha = {
+...                     '%s_%s' % (alpha_metrics[met], dep):
+...                     agu.get_existing_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['source-%s-%s' % (dep, met)])
+...                     for (met, dep) in itertools.product(alpha_metrics.keys(), depth_names[:(depth_id+1)])
+...                 }
+...
+...                 source_beta = [
+...                     agu.get_existing_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['source-%s-unweighted-unifrac' % depth]),
+...                     agu.get_existing_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['source-%s-weighted-unifrac' % depth]),
+...                 ]
+...
+...                 sink_dir = agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-dir' % depth])
+...                 sink_otu = agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-otu' % depth])
+...                 sink_map = agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-map' % depth])
+...                 sink_beta = [
+...                     agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-unweighted-unifrac' % depth]),
+...                     agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-weighted-unifrac' % depth]),
+...                     ]
+...
+...                 yield (bodysite, depth, source_otu, source_map, source_alpha, source_beta, sink_dir, sink_otu, sink_map, sink_beta)
+```
+
+Let's use our helper function to add the alpha diversity to our mapping file, and move the rest of the data we need.
+
+```python
+>>> for (bodysite, depth, source_otu, source_map, source_alpha, source_beta, sink_dir, sink_otu, sink_map, sink_beta) in rarefied_parameter_iterator():
```

Comment:
Is there documentation elsewhere that explains what sinks and sources are?  If not, it may be preferable to have an explanation here.